### PR TITLE
chore: get calico and v-k from mcr.microsoft.com

### PR DESCRIPTION
**Pull request by mboersma**
**Reason for Change**:

Moves references to the calico and virtual-kubelet components to mcr.microsoft.com, where they are already published.

**Issue Fixed**:

Refs #3753

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

I `docker pull`ed all these images to make sure they're actually published to MCR.

In Mooncake, AKS Engine uses `dockerhub.azk8s.cn/calico/` as the base for Calico images and `dockerhub.azk8s.cn/microsoft/` as the base for virtual-kubelet. Are there parallel replacements at `mcr.azk8s.cn`?

### CODE REVIEW

**Review by jackfrancis — APPROVED** [review]
/lgtm

thanks @mboersma!

File: vhd/packer/install-dependencies.sh [hunk 1]
```diff
@@ -285,7 +285,7 @@ done
 
 VIRTUAL_KUBELET_VERSIONS="latest"
 for VIRTUAL_KUBELET_VERSION in ${VIRTUAL_KUBELET_VERSIONS}; do
-    CONTAINER_IMAGE="microsoft/virtual-kubelet:${VIRTUAL_KUBELET_VERSION}"
+    CONTAINER_IMAGE="mcr.microsoft.com/oss/virtual-kubelet/virtual-kubelet:${VIRTUAL_KUBELET_VERSION}"
```
Comment (sozercan) [inline vhd/packer/install-dependencies.sh:288]: can we pin this? cc @cpuguy83

Comment (mboersma) [reply vhd/packer/install-dependencies.sh:288]: It's always floated at `:latest`, but we could use `:0.9.1`. They point to the [same image](https://hub.docker.com/r/microsoft/virtual-kubelet/tags) and haven't been updated in a year.

Comment (cpuguy83) [reply vhd/packer/install-dependencies.sh:288]: The hub image is not able to be updated anymore (per policy), which is why it was moved to ACR.
Honestly, I think the virtual-node add-on is using 'latest' as well 😞

Comment (sozercan) [reply vhd/packer/install-dependencies.sh:288]: Looks like `microsoft/virtual-kubelet` registry is deprecated. published `v1.2.1.2` to MCR.

Comment (mboersma) [reply vhd/packer/install-dependencies.sh:288]: I think pinning virtual-kubelet to a recent version is logically separate from this PR, so I'm going to merge this and we can follow up on #3810.

**Review by sozercan — APPROVED** [review]

**Pull request merged**